podman/api/client.py

```diff
@@ -178,7 +178,7 @@ def _normalize_url(base_url: str) -> urllib.parse.ParseResult:
     def delete(
         self,
         path: Union[str, bytes],
-        params: Union[None, bytes, Mapping[str, str]] = None,
+        params: Union[None, bytes, Mapping[str, Any]] = None,
```

Comment:
This change seems spurious as none of the other like methods were flagged, ie get/put/head/... These methods all call _request() which has `params: Union[None, bytes, Mapping[str, str]] = None,`

Reply:
This change was to fix this error:
```
podman/domain/networks_manager.py:202: error: Dict entry 0 has incompatible type "str": "bool | None"; expected "str": "str"  [dict-item]
```
This is the call to delete in that file:
```
response = self.client.delete(f"/networks/{name}", params={"force": force})
```
Where force is typed `Optional[bool]`.

Reply:
So the type checker didn't flag the delete method directly; it flagged the call to it which did not follow the type hints.

Reply:
Ah, should the caller be fixed instead?

Reply:
I think it is reasonable to call `client.delete("...", params={"force": True}` so supporting a bool value should be good.

However, I'm not sure the use of `Mapping[str, Any]` is the best approach. Since the value for params is sent directly to `requests.Session.request`, the most appropriate supported type would be that one. [Typeshed defines it as](https://github.com/python/typeshed/blob/7646d38302ab7c2d56c2620ac43928331cc65c98/stubs/requests/requests/sessions.pyi#L95-L103):
```python
_ParamsMappingKeyType: TypeAlias = str | bytes | int | float
_ParamsMappingValueType: TypeAlias = str | bytes | int | float | Iterable[str | bytes | int | float] | None
_Params: TypeAlias = (
    SupportsItems[_ParamsMappingKeyType, _ParamsMappingValueType]
    | tuple[_ParamsMappingKeyType, _ParamsMappingValueType]
    | Iterable[tuple[_ParamsMappingKeyType, _ParamsMappingValueType]]
    | str
    | bytes
)
```

But this seems like a lot. Also, if I understand correctly, using [Any](https://docs.python.org/3/library/typing.html#the-any-type) disables type verification for the specific variable.

Maybe we can just add support for bool to allow for the specific usage and adapt later as required?

```python
params: Union[None, bytes, Mapping[str, Union[str, bool]]] = None,
```

Reply:
I am comfortable with that change, as long as it is propagated to all the http verb helper functions, ie get/post/etc